### ISSUE TITLE
TTS output very poor on custom data

### ISSUE THREAD

**Issue by samin9796** [espnet/espnet#4831]
Hello! Previously, I trained Tacotron 2 and Fast Speech 2 models on two hours of single speaker (female) custom dataset following LJSpeech recipe (just an exception that I increased the accum_grad value since I did not have enough GPU memory). The outputs were very good then. Now, I am trying to train a Tacotron 2 model on a new dataset of the same language following the same procedure. This time it has 4.5 hours of speech data collected from a single male speaker. One more difference is that now the average length of each utterance is about 20 second (max duration: 29 sec, min duration: 2.8 sec), whereas in the previous dataset, each utterances were only of 3-5 seconds. The current model outputs are really poor. I have tried out two accumulation gradient values (e.g. 2 and 4).

Here are the attention weights for the Tacotron 2 model with accum_grad=2 and batch_bins=4000000:
![Alt Text](https://media.giphy.com/media/1ezUMyNWV319sTWISp/giphy.gif)

Here are the attention weights for Tacotron 2 model with accum_grad=4 and batch_bins=4000000:
![Alt Text](https://media.giphy.com/media/ewRKTl8LQBgPIZWMSU/giphy.gif)

Using accum_grad=4, I could see some diagonal shape upto 37 epochs but it disappeared from epoch 38. I am always using characters as tokens since there is no lexicon/G2P converter available for the language I am working with.

In this case, I have the following questions:

1) Could you please tell me what might be going wrong here? 
2) Can the longer audio duration of the new dataset be reason of worse performance? 
3) If I want to train a FastSpeech 2 model using my new data, is it possible to use the previous Tacotron 2 model (trained on same language but different speaker) to get the durations?

I would really appreciate any solution!

**Comment by kan-bayashi** [comment]
Most of the cases, the cleaning of training data is critical.

See tips:
[master/egs2/TEMPLATE/tts1#why-the-model-cannot-be-trained-well-with-my-own-dataset](https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/tts1?rgh-link-date=2021-04-18T11%3A41%3A43Z#why-the-model-cannot-be-trained-well-with-my-own-dataset)

Related issue:
https://github.com/espnet/espnet/issues/2327
https://github.com/espnet/espnet/issues/2565

**Comment by samin9796** [comment]
Thank you for the tips. I split up the long audio files into multiple smaller ones using forced alignment technique and removed long silences. Now I am getting reasonably good outputs.

**Comment by Curiosci** [comment]
How long were the audio files that you splitted ?

**Comment by samin9796** [comment]
@Curiosci The audio files were initially on average 19 seconds long.

**Comment by Curiosci** [comment]
And what's the average duration that you considered after splitting ?

**Comment by samin9796** [comment]
On average approx. 5 seconds

**Comment by Curiosci** [comment]
How about silences? what threshold did you use?

**Comment by samin9796** [comment]
Can't remember the threshold. But long silences should be truncated. Don't truncate too much since it might affect the naturalness of speech.

**Comment by Curiosci** [comment]
@samin9796 Thanks for the replies!
